modules/product/src/main/java/com/opengamma/strata/product/SecurityProduct.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.market.ReferenceData;
+
+/**
+ * The product details of a security.
+ * <p>
+ * A {@code SecurityProduct} includes all the financial details of a security.
+ * It will include any underlying securities.
+ * For example, a bond future option would include the underlying bond future
+ * and the basket of bonds.
+ * <p>
+ * The product of a security is distinct from the {@linkplain ReferenceSecurity security}.
+ * The security, referred to using a {@link SecurityId}, does not include underlying securities.
+ * <p>
+ * Implementations of this interface must be immutable beans.
+ */
+public interface SecurityProduct
```

Comment:
Why do we need `ReferenceSecurity` and `SecurityProduct`? There is one more level of indirection than I was expecting. In my mental model an equity trade contains a security ID and the ID refers to an equity (the thing that is currently called `EquityProduct`).

If I'm understanding this model correctly, the trade refers to a security ID, the ID refers to a `ReferenceSecurity` and the security refers to a `Product` implementation. Could `EquitySecurity` and `Equity` be collapsed into a single entity which is referenced by security ID and stored in the reference data?